spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveProcedures.scala

```diff
@@ -50,8 +50,9 @@ case class ResolveProcedures(spark: SparkSession) extends Rule[LogicalPlan] with
 
     // optional params should be at the end
     procedure.parameters.sliding(2).foreach {
-      case Array(previousParam, currentParam) if previousParam.required && !currentParam.required =>
```

Comment:
This was a bug I overlooked.

Reply:
Should this be in a separate PR with a test?

Reply:
Done.